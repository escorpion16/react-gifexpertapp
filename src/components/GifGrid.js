import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const {data: images,loading} = useFetchGifs(category);

    return (
        <>
            <h3 className='card animate__animated animate__fadeIn'> { category } </h3>
            { loading && 'Loading...' }
            <div className="wrapperCards ">
                {
                    images.map( img => (
                        <GifGridItem key={ img.id } {...img} />
                    ))
                } 
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}