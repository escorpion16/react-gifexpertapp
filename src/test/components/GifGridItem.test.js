import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas de <GifGridItem />', () => {
    
    const title = 'es un titulo';
    const url = 'https://localhost/imagen.jpg';
    const wrapper = shallow( <GifGridItem url={ url } title={ title } /> )

    test('debe de mostrar el componente correctamente', () => {

        expect( wrapper ).toMatchSnapshot()
    })

    test('debe de ser igual al title recibido', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title )
    })

    test('debe ser una url de imagen correcta', () => {
        const image = wrapper.find('img');

        expect( image.prop('src') ).toBe( url )
        expect( image.prop('alt') ).toBe( title )
    })

    test('debe de contener animate__fadeInLeft', () => {
        const div = wrapper.find('div');
        const className = div.prop('className')
        expect( className.includes('animate__fadeInLeft') ).toBe(true)
    })
})

