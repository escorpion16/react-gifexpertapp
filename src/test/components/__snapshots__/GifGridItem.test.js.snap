// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img
    alt="es un titulo"
    src="https://localhost/imagen.jpg"
  />
  <p>
     
    es un titulo
     
  </p>
</div>
`;
